Core/FunctionTree/FunctionTreeIntensityWrapper.cpp

```diff
@@ -0,0 +1,89 @@
+#include "FunctionTreeIntensityWrapper.hpp"
+#include "Core/Kinematics.hpp"
+#include "FunctionTree.hpp"
+#include "Intensity.hpp"
+#include "Value.hpp"
+
+namespace ComPWA {
+namespace FunctionTree {
+/*FunctionTreeIntensityWrapper::FunctionTreeIntensityWrapper(
+    std::shared_ptr<FunctionTree> tree, ParameterList parameters,
+    ParameterList data)
+    : Tree(tree), Parameters(parameters), Data(data) {}*/
+
+FunctionTreeIntensityWrapper::FunctionTreeIntensityWrapper(
+    std::shared_ptr<OldIntensity> oldintens, size_t VariableCount,
+    std::string name)
+    : OldIntens(oldintens) {
+  for (size_t i = 0; i < VariableCount; ++i) {
+    std::vector<double> temp;
+    Data.addValue(std::make_shared<Value<std::vector<double>>>(temp));
+  }
+  Tree = oldintens->createFunctionTree(Data, name),
+  oldintens->addUniqueParametersTo(Parameters);
+  UserParameters.DeepCopy(Parameters);
+  for (auto x : Parameters.doubleParameters()) {
+    x->fixParameter(false);
+    // IMPORTANT: we have to unfix all parameters
+    // since the optimizer will take care of this
+    // and we can't maintain which parameters will
+    // be fixed later on. So internally we treat them
+    // all as free. But of course only the parameters
+    // that are not fixed will change...
+    // So the function tree caching and recalculation
+    // still works fine
+    // When improving the FunctionTree later on
+    // The Parameters should be more "dumb" and not have
+    // information about a fixed or not fixed status.
+  }
+}
+
+FunctionTreeIntensityWrapper::FunctionTreeIntensityWrapper(
+    std::shared_ptr<OldIntensity> oldintens,
+    std::shared_ptr<ComPWA::Kinematics> kin, std::string name)
+    : FunctionTreeIntensityWrapper(
+          oldintens, kin->getKinematicVariableNames().size(), name) {}
+
+std::vector<double> FunctionTreeIntensityWrapper::evaluate(
+    const std::vector<std::vector<double>> &data) {
+  updateDataContainers(data);
+  Tree->UpdateAll(Tree->head());
```

Comment:
Does that mean that the `FunctionTree` is recalculated completely each time the data vectors have changed?

Reply:
Yeah there are better ways, but this was the easiest change for me to make it work. I would put improvements into this direction into the FuctionTree improvement project. Note that this has no influence on the fitting performance since this only occurs in the intensity and not the estimator

Reply:
But the effect on generating data is huge, since also the normalization terms (which are constant after data has changed) are recalculated. Are you sure that with the current FunctionTree implementation is recalculation is not done properly?

Reply:
Thats true. I will try to do it correctly.

Reply:
Ok just adding a `Notify()` in `Value::setValue()` analogous to `FitParameter::setValue()` worked! So this should be solved as well